parameterized/parameterized.py

```diff
@@ -386,6 +386,36 @@ def check_input_values(cls, input_values):
             input_values = list(input_values)
         return [ param.from_decorator(p) for p in input_values ]
 
+
+    @classmethod
+    def parameterized_class(cls, attribute_names, input_values):
+        """
+        A method for parameterizing test classes.
+        This sets the defined tuples in properties as attributes in the class
+        and sets its corresponding values defined in input_values
+        :param attribute_names: tuple of strings or string value
+        :param input_values: array of tuples
+
+        """
+        def decorator(base_class):
+            test_class_module = sys.modules[base_class.__module__].__dict__
+            for test_value_key, test_field in enumerate(input_values):
+                test_class_dict = dict(base_class.__dict__)
+
+                if isinstance(attribute_names, string_types):
+                    test_class_dict[attribute_names] = test_field
+                elif len(attribute_names) == len(test_field):
+                    for j, property_key in enumerate(attribute_names):
+                        test_class_dict[property_key] = test_field[j]
+                _create_module(base_class, test_class_module, test_value_key, test_class_dict)
+
+        def _create_module(b, test_class_module, test_value_key, test_class_dict):
+            name = '{method_name}_{index}'.format(method_name=b.__name__, index=test_value_key + 1)
```

Comment:
I think it would be nice if it follows up the existing convention which string param value appends to new test method name
ex)
```python
name = name_func(f, num, p)

# as result, new test method name would be like
test_api_0_v1 (test.APITestCase) ... ok
test_api_1_v2 (test.APITestCase) ... ok
test_api_2_v3 (test.APITestCase) ... ok
```

I will try to make commits on you pr after work but it will be probably at least in 8 hours cuz I'm in KST(UTC +9).
By the way great work @charlsagente!!

Reply:
Thank you very much @legshort, take your time.

Reply:
+1 - that's definitely the nicest way to do it.